components/App.jsx

```diff
@@ -47,14 +49,15 @@ const App = () => {
                 <GlobalStyle />
                 <AudioButton />
 
-                {isNav && <MainNav setIsNav={setIsNav} setIsSpeechBoxActive={setIsSpeechBoxActive} />}
+                {isNav && <MainNav setIsNav={setIsNav} isNav={isNav} setIsSpeechBoxActive={setIsSpeechBoxActive} />}
 
                 <Sprite
                     isSpriteInSpeechBox={isSpriteInSpeechBox}
                     spriteClicked={spriteClicked}
                     spriteClickHandler={spriteClickHandler}
                     isInitialSpriteClick={isInitialSpriteClick}
                     isSpriteNavButton={isSpriteNavButton}
+                    location={location}
```

Comment:
Ok, we definitely want location in a useEffect. 

It looks like you're expecting location to update as the URL changes. We want to control that state so we aren't grabbing a new location every time any state change happens anywhere in the app. useEffect will allow us to control the updating of location to only occur when there is a diff between our current location and the most up to date one